### PR TITLE
hooks: add metrics

### PR DESCRIPTION
**Pull request by gsquared94**
This PR adds instrumentation for `lifecycle hooks` usage. It collects the number of hooks configured per phase.

### CODE REVIEW

File: pkg/skaffold/instrumentation/export.go [hunk 1]
```diff
@@ -167,6 +167,8 @@ func createMetrics(ctx context.Context, meter skaffoldMeter) {
 	m := global.Meter("skaffold")
 
 	// cloud monitoring only supports string type labels
+	// cloud monitoring only supports 10 labels per metric descriptor
+	// be careful when appending new values to this `labels` slice
```
Comment (MarlonGamez) [inline pkg/skaffold/instrumentation/export.go:171]: do we track how many labels we are using so far and what they're being used for?

**Review by MarlonGamez — APPROVED** [review]
LGTM 👍🏼

File: pkg/skaffold/instrumentation/export.go (hunk 1 above)
Comment (gsquared94) [reply pkg/skaffold/instrumentation/export.go:171]: I meant this as a comment for the immediate `labels` slice on line 172

Comment (gsquared94) [reply pkg/skaffold/instrumentation/export.go:171]: but you're right, we don't have any document for the exact set across all metrics

**Pull request merged**